src/blockdata/transaction.rs

```diff
@@ -332,19 +325,50 @@ impl Transaction {
     /// - Does NOT attempt to support OP_CODESEPARATOR. In general this would require evaluating
     /// `script_pubkey` to determine which separators get evaluated and which don't, which we don't
     /// have the information to determine.
-    /// - Does NOT handle the sighash single bug, you should either handle that manually or use
-    /// [`Self::signature_hash()`] instead.
+    /// - Does NOT handle the sighash single bug (see "`Option` return value" section)
+    ///
+    /// # `Option` return value
+    ///
+    /// This function can't handle the SIGHASH_SINGLE bug internally. However if it was
+    /// to return an error in this case, a lot of users would mishandle it by calling
+    /// `.expect()` on the result, assuming that `writer` they have passed can't possibly
+    /// fail unaware of this additional case. For this reason the `Result` is wrapped
+    /// in an `Option`. In the case of SIGHASH_SIGNLE bug, the function will return
+    /// `None` without writting anything to the `writer`.
+    ///
+    /// Caller is recommended to use the following pattern to handle this corner case:
+    ///
+    /// ```rust
+    /// # use bitcoin::consensus::deserialize;
+    /// # use bitcoin::{Transaction, Sighash};
+    /// # use bitcoin_hashes::{Hash, hex::FromHex};
+    /// # let mut writer = Sighash::engine();
+    /// # let input_index = 0;
+    /// # let script_pubkey = bitcoin::Script::new();
+    /// # let sighash_u32 = 0u32;
+    /// # const SOME_TX: &'static str = "0100000001a15d57094aa7a21a28cb20b59aab8fc7d1149a3bdbcddba9c622e4f5f6a99ece010000006c493046022100f93bb0e7d8db7bd46e40132d1f8242026e045f03a0efe71bbb8e3f475e970d790221009337cd7f1f929f00cc6ff01f03729b069a7c21b59b1736ddfee5db5946c5da8c0121033b9b137ee87d5a812d6f506efdd37f0affa7ffc310711c06c7f3e097c9447c52ffffffff0100e1f505000000001976a9140389035a9225b3839e2bbf32d826a1e222031fd888ac00000000";
+    /// # let raw_tx = Vec::from_hex(SOME_TX).unwrap();
+    /// # let tx: Transaction = deserialize(&raw_tx).unwrap();
+    /// if let None = tx.encode_signing_data_to(&mut writer, input_index, &script_pubkey, sighash_u32)
+    ///         .transpose()
+    ///         .expect("writer doesn't error") {
```

Comment:
In 9435eda1077743065a9e22b031d21275a5ab7f34:

`.is_none()` is more idiomatic than `if let None =`

Reply:
Ha. :facepalm: :+1: